tests/services/users/test_service_users.py

```diff
@@ -59,8 +59,8 @@ def test_search_public_users(user_service, user_pub):
     [
         "email:res@inveniosoftware.org",
         "res@inveniosoftware.org",
-        "email:pubres@inveniosoftware.org",
-        "pubres@inveniosoftware.org",
+        'email:"pubres@inveniosoftware.org"',
```

Comment:
sorry, just for me to understand the implications (correct me if I am saying something wrong):
the `""` enforces exact match for the phrase being searched, correct? wasn't it the whole point **not to** force the user to use the double quotes in the search phrases, since most of the people does not know about it?

Reply:
if you search for these without the quotes, it returns the user with email `pub@inveniosoftware.org` as there is a partial match. The test is looking for there to be no matches for the search query, and so doing an exact match causes the search to return no results.

The user will still be able to search for things without the quotes :)

Reply:
maybe using email tokenizer would help with this also
https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-uaxurlemail-tokenizer.html

Reply:
oh! it's actually matching on the `inveniosoftware.org`

```
 curl -X POST "http://localhost:9200/_analyze?pretty=true" -H "Content-Type: application/json" -d '{
  "text": "email:pubres@inveniosoftware.org"
}'
{
  "tokens" : [
    {
      "token" : "email:pubres",
      "start_offset" : 0,
      "end_offset" : 12,
      "type" : "<ALPHANUM>",
      "position" : 0
    },
    {
      "token" : "inveniosoftware.org",
      "start_offset" : 13,
      "end_offset" : 32,
      "type" : "<ALPHANUM>",
      "position" : 1
    }
  ]
}
```

as the default search tokenizer splits on the `@`. Which essentially means that it will always do a domain search 🤷 We would have to change the way we tokenize search queries to avoid this

Reply:
I think the email tokenizer would help with the domain match, no?

Reply:
It is working better with the email tokenizer and I now don't need to change the tests at all :)